### DISCUSSION TITLE
Error when trying to start element-wait plugin

### DISCUSSION THREAD

**Discussion by LizGrillo** [appium/appium#19732] [Q&A] [answered]
Hi everyone,

I need some help with the wait plugin, please. 

So I have the element-wait plugin installed and Im trying to start it. I'm getting this error: 

lizzygrillo@Lizzies-MacBook-Pro-2 AndroidTastyAppium2 %  appium --use-plugins=element-wait
[Appium] Attempting to load plugin element-wait...
[Appium] Requiring plugin at /Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/index.js
[Appium] **Could not load plugin 'element-wait', so it will not be available. Error in loading the plugin was: Cannot find module '@appium/base-plugin'**
[Appium] Require stack:
[Appium] - /Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/plugin.js
[Appium] - /Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/index.js
[Appium] Error: Cannot find module '@appium/base-plugin'
[Appium] Require stack:
[Appium] - /Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/plugin.js
[Appium] - /Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/index.js
[Appium]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1090:15)
[Appium]     at Function.Module._load (node:internal/modules/cjs/loader:934:27)
[Appium]     at Module.require (node:internal/modules/cjs/loader:1157:19)
[Appium]     at require (node:internal/modules/helpers:119:18)
[Appium]     at Object.<anonymous> (/Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/plugin.js:15:39)
[Appium]     at Module._compile (node:internal/modules/cjs/loader:1275:14)
[Appium]     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1329:10)
[Appium]     at Module.load (node:internal/modules/cjs/loader:1133:32)
[Appium]     at Function.Module._load (node:internal/modules/cjs/loader:972:12)
[Appium]     at Module.require (node:internal/modules/cjs/loader:1157:19)
[Appium]     at require (node:internal/modules/helpers:119:18)
[Appium]     at Object.<anonymous> (/Users/lizzygrillo/.appium/node_modules/appium-wait-plugin/lib/index.js:7:34)
[Appium]     at Module._compile (node:internal/modules/cjs/loader:1275:14)
[Appium]     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1329:10)
[Appium]     at Module.load (node:internal/modules/cjs/loader:1133:32)
[Appium]     at Function.Module._load (node:internal/modules/cjs/loader:972:12)
[Appium] Welcome to Appium v2.4.1 (REV 552db40622bb7a82d9c6d67d2d6bcf3694b47e30)
[Appium] Non-default server args:
[Appium] {
[Appium]   usePlugins: [
[Appium]     'element-wait'
[Appium]   ]
[Appium] }
[Appium] The autodetected Appium home path: /Users/lizzygrillo/.appium
[Appium] Attempting to load driver uiautomator2...
[Appium] Attempting to load driver xcuitest...
[Appium] Requiring driver at /Users/lizzygrillo/.appium/node_modules/appium-xcuitest-driver/build/index.js
[Appium] XCUITestDriver has been successfully loaded in 1.403s
[Appium] Requiring driver at /Users/lizzygrillo/.appium/node_modules/appium-uiautomator2-driver/build/index.js
[Appium] AndroidUiautomator2Driver has been successfully loaded in 2.790s
[Appium] Appium REST http interface listener started on http://0.0.0.0:4723
[Appium] You can provide the following URLs in your client code to connect to this server:
[Appium]        http://127.0.0.1:4723/ (only accessible from the same host)
[Appium]        http://192.168.0.197:4723/
[Appium] Available drivers:
[Appium]   - uiautomator2@2.43.1 (automationName 'UiAutomator2')
[Appium]   - xcuitest@5.14.1 (automationName 'XCUITest')
[Appium] Available plugins:
[Appium]   - element-wait@3.0.1
[Appium]   - appium-dashboard@v2.0.3
[Appium]   - gestures@4.0.0
[Appium]   - appium-reporter-plugin@1.1.0-beta.05
[Appium] No plugins activated. Use the --use-plugins flag with names of plugins to activate

I don't know what to do. Can someone help please? If so, could you please tell me what to do in a detailed way since Im not a senior? Sometimes it's hard to follow the steps for me because they are very technical

Same problem when trying to activate the gesture plugin.It seems I have problems with all my plugins. I have them installed but It's not possible for me to activate them

**Comment by jlipps** [comment]
This looks like an error in the plugin itself, which is not maintained by the Appium team. You should contact the plugin authors and report this as a bug to them.

**Reply by LizGrillo** [reply]
Ok, thanks

**Comment by LizGrillo** [comment, marked as answer]
So, for those who ever encounter this problem you need to install the @appium/base-plugin: 

npm install @appium/base-plugin

Then it will work.
